api.py

```diff
@@ -64,6 +64,28 @@ def list_templates(**kwargs):
 
 
 def wait_for_job(job_id, **kwargs):
+    """
+    Wait for a job to complete.
+
+    Block execution until the completion of an asynchronous job.
+
+    Parameters
+    ----------
+    job_id : string
+        The UUID that identifies the job to query.  The job ID is returned in the response header of the request that starts the job.
+    **kwargs:
```

Comment:
What makes `ConduceCreds` different from `kwargs`?

Reply:
`ConduceCreds` would be an object with only the parameters required to get onto a specific server, like host & api-key, or host & email.  Flags like `wipe`, `create-accounts`, `orch` and `acl` couldn't hitch a ride

Reply:
What's the harm if they do?

Reply:
I have a few reasons.  As having used docs for python libs that use kwargs, it's prone to error and leads to inconsistency over time.

When I use these libs it's frustrating as a user because kwargs is just a vague bucket.  I have to go hunting for info. Oftentimes I have variables I want to pass in, like login info, and just want to see parameter names since I can figure out what they mean (I want to delete something, and I see remove, that must be what I want to use - or is it email? user? account? - or is it host? url? env? server?).  I also want a low friction way to see all the parameters I can use to change behaviour.  For example `help(json.dumps)` shows `dumps(obj, skipkeys=False, ensure_ascii=True, check_circular=True, allow_nan=True, cls=None, indent=None, separators=None, encoding='utf-8', default=None, sort_keys=False, **kw)` as the signature.  The names are descriptive and it makes it really easy to see what sort of knobs I can tweak.  There is a `**kw` in there, and finding more info on that is not something I want to do.  Reading the docstring to find out is enough work, I don't want to go looking in more places.

I also not-uncommly find errors with `**kwargs` doc strings too.  Code changes and the docs get out of date.  Someone updates the kwargs that get used, but forgets to update the docs.  If kwargs are explicitly specified in the function signature that's always up to date.

Additionally, it is easy to mess things up.  I can have a typo in a `**kwarg` key and then all bets are off - sometimes `**kwargs` is written to use defaults, sometimes it crashes.  If I a kwarg is explicitly specified, and I have a typo, I will get a `TypeError got an unexpected keyword argument 'mytypo'`.  Along the same lines, if I pass in a blob of kwargs into a function, I might get unlucky & have a collision - the `orch` flag is True b/c I want to load the orch yaml config, but maybe a permissions function will share an orch if its id is given under `kwargs[orch]` & then I hit an error because I tried to share an orch with an ID of `True`.

From a maintainer perspective, I'd say the previous point is a big one & encapsulation is important - it's easy to see what info is used where.  It'd be easy to make observations like, this command line args go into the credentials which are passed around to anything that uses the API.  The create-accounts flag is only used when building the teams, etc.